### ISSUE TITLE
Deconstruction-declaration with pointer types fails to parse

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#48963]
```
using System;
public class C 
{
    public unsafe void M() 
    {
        (int* x, int*[] y) = this;
    }
    
    public unsafe void Deconstruct(out int* x, out int*[] y) => throw null;
}
```
[sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDZICYgNQA+AAgEwCMAsAFBEDMABKfQML3UDe193jDArgDsoEAGZxGAFnoBZABQBKNlR71OylT1loBMAFT0AHknra9AbQC69AJ6KAvPRgALNFADcXHgF9P3X73pBYTFJegAROABjAHshGAQ+SJhZaL4YEx19I3pU9NNdSxt7AD5HJwRogHd6AT4MDA8qLyA)

**Comment by gafter** [comment]
This is an error because the tuple type on the right contains two pointers, but tuples are not permitted to contain pointers.

**Comment by jcouv** [comment]
@gafter Right, but that's not the problem I was trying to point out. We're getting a parsing error.
I've updated the OP snippet to remove the distraction.

**Comment by Darcara** [comment]
I've run into this problem too. Using `var` instead of `int*` parses and compiles fine.
Declaring the variables before the deconstruct call works too.
All 3 notations seem to produce the same IL with no actual tuples involved.
